### PR TITLE
Figures for overleaf

### PR DESCRIPTION
**Pull request by JonathanHillman**

### CODE REVIEW

File: README.org [hunk 1]
```diff
@@ -11,7 +11,7 @@ repositories: [[https://github.com/tdhock/feature-learning-benchmark]] and
   [[file:figure-aum-convexity.png][right PNG (AUM/AUC functions)]], [[file:figure-aum-convexity.R][R code]].
 - Figure 4: optimizing AUM maximizes AUC. [[file:figure-aum-optimized-iterations.png][Left PNG]], [[file:figure-aum-optimized.R][R code]]
   (iterations); [[file:figure-aum-train-both.png][Right PNG]], [[file:figure-aum-train-both.R][R code]] (ROC curves).
-- Figure 5: TODO Jon.
+- Figure 5: Test AUM/AUC comparison across multiple seeds and datasets [[file:figure-test-comparison.R][R code]]
```
Comment (tdhock) [inline README.org:14]: @JonathanHillman please add link to PNG files

File: figure-test-comparison.R [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                  file.path("neuroblastoma-data/data/H3K4me3_XJ_immune/cv/equal_labels/testFolds/2"),
+                  file.path("neuroblastoma-data/data/H3K4me3_XJ_immune/cv/equal_labels/testFolds/4"))
+
+#png.testFold.vec <- sapply(testFold.vec, function(x){file.path(paste0(substring(x, 1, nchar(x) - 4), ".png"))})
+
+
+
+# lapply(1:3, function(x){pdf_convert(testFold.vec[x], png.testFold.vec[x] )})
+# for(index in 1:length(testFold.vec))
+# {
```
Comment (tdhock) [inline figure-test-comparison.R:21]: please remove commented code

File: figure-test-comparison.R [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+library(ggplot2)
+library(data.table)
+
+# testFold.vec <- Sys.glob("../AUM meeting figures/28Jan2021/*/linear-model-post-cv-aum.csv")
+# precv.vec <- Sys.glob("../AUM meeting figures/28Jan2021/*/linear-model-cv-aum.csv")
+# postcv.vec <- Sys.glob("../AUM meeting figures/28Jan2021/*/linear-model-post-cv-aum.csv")
+
+testFold.vec <- c(file.path("neuroblastoma-data/data/ATAC_JV_adipose/cv/equal_labels/testFolds/4"),
+                  file.path("neuroblastoma-data/data/H3K27ac-H3K4me3_TDHAM_BP/cv/equal_labels/testFolds/2"),
+                  file.path("neuroblastoma-data/data/systematic/cv/R-3.6.0-profileSize/testFolds/1"),
```
Comment (tdhock) [inline figure-test-comparison.R:11]: please remove repetition, file.path(neuroblastoma-data/data

File: packages.R [hunk 4]
```diff
@@ -71,6 +71,8 @@ works_with_R(
   gridExtra="2.3",
   neuroblastoma="1.0",
   tikzDevice="0.12.3.1",
+  microbenchmark="1.4.7",
+  directlabels="2021.2.24",
```
Comment (tdhock) [inline packages.R:75]: please remove this directlabels line, use the tdhock/directlabels@hash line above instead (more recent)

File: figure-test-comparison.R [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                                                                                auc.iteration = min.aum.iteration)
+          
+          selected.auc.dt.list[[paste0(curr.seed, curr.cv.algo)]] <- data.table(`Test AUM` = curr.dt$aum[max.auc.iteration],
+                                                                                `Test AUC` = curr.dt$auc[max.auc.iteration],
+                                                                                algorithm = paste0("Max.Valid.AUC"),
+                                                                                seed = curr.seed,
+                                                                                aum.iteration = max.auc.iteration,
+                                                                                auc.iteration = max.auc.iteration)
+        }
+      }
```
Comment (tdhock) [inline figure-test-comparison.R:105]: use for loop over auc/aum instead of block of code repeated for each

File: figure-test-comparison.R [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+      #         geom_point(aes(x = aum, y = type, color = seed)) +
+      #         ggtitle(plot.title))
+      # 
+      # auc.list[[curr.index]] <- (ggplot(data = test.aum.dt) +
+      #         geom_point(aes(x = auc, y = type, color = seed)) +
+      #         ggtitle(c(data.name, cv.type, test.fold)))
+}
+
+test.aum.dt.combined <- do.call(rbind, test.aum.dt.list) %>%
+  mutate(new.test.fold = paste0("Test Fold ", test.fold))
```
Comment (tdhock) [inline figure-test-comparison.R:141]: instead of this can you have "Test Fold 2, n=500, p=34"

File: figure-test-comparison.R [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+png("figure-test-auc-comparison.png", width = 26, height = 3, res = 200, units = "in")
+
+
+ggplot(data = test.aum.dt.combined) +
+  geom_point(aes(x = `Test AUC`, y = algorithm, color = seed), size = 5) +
```
Comment (tdhock) [inline figure-test-comparison.R:147]: please remove seed legend

**Comment by tdhock** [comment]
why close? I would like to eventually merge your code for reproducibility purposes.

**Comment by JonathanHillman** [comment]
I deleted this pull request because its from a fork, and I created a new one from my branch instead

**Pull request closed without merging**